lib/compute/zone.js

```diff
@@ -369,6 +375,79 @@ Zone.prototype.createDisk = function(name, config, callback) {
 };
 
 /**
+ * Create an instance group in this zone.
+ *
+ * @resource [InstanceGroup Resource]{@link https://cloud.google.com/compute/docs/reference/v1/instanceGroups#resource}
+ * @resource [InstanceGroups: insert API Documentation]{@link https://cloud.google.com/compute/docs/reference/v1/instanceGroups/insert}
+ *
+ * @param {string} name - Name of the instance group.
+ * @param {object} options - See an
+ *     [InstanceGroup resource](https://cloud.google.com/compute/docs/reference/v1/instanceGroups#resource).
+ * @param {object} options.ports - A map of names to ports. The key should be
+ *     the name, and the value the port number. Maps to `options.namedPorts`.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {module:compute/instance-group} callback.instanceGroup - The created
+ *     InstanceGroup object.
+ * @param {module:compute/operation} callback.operation - An operation object
+ *     that can be used to check the status of the request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * function onCreated(err, instanceGroup, operation, apiResponse) {
+ *   // `instanceGroup` is an InstanceGroup object.
+ *
+ *   // `operation` is an Operation object that can be used to check the status
+ *   // of the request.
+ * }
+ *
+ * zone.createInstanceGroup('instance-group-name', onCreated);
+ */
+Zone.prototype.createInstanceGroup = function(name, options, callback) {
+  var self = this;
+
+  if (is.fn(options)) {
+    callback = options;
+    options = {};
+  }
+
+  var body = extend({}, options, {
+    name: name
+  });
+
+  if (body.ports) {
+    body.namedPorts = [];
+
+    for (var namedPort in body.ports) {
```

Comment:
Since this logic is present elsewhere, maybe we could refactor it into a private static method?